test/dialect/postgresql/test_dialect.py

```diff
@@ -373,6 +462,29 @@ def test_connect_psycopg_multiple_hosts(self, pattern):
         with e.connect() as conn:
             eq_(conn.exec_driver_sql("select 1").scalar(), 1)
 
+    @testing.only_on(["+asyncpg"])
+    @testing.combinations(
+        "host=H:P&host=H:P&host=H:P",
+        "host=H,H,H&port=P,P,P",
+    )
+    def test_connect_asyncpg_multiple_hosts(self, pattern):
```

Comment:
**mike bayer** (zzzeek) wrote:

what does this test that the test_multi_asyncpg_hosts method does not test?  there should be just one set of tests for this if nothing is new here

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4755

Reply:
It tests a connection, not just parsing a dsn string arguments, like `test_multi_asyncpg_hosts`
I've copied it from `test_connect_psycopg_multiple_hosts` and adopted it for testing positive scenarios of asyncpg connection.

Reply:
OK why only asyncpg then?

Reply:
this is separate test for asyncpg only is because asyncpg have different positive scenarios (it doesn't set default port when port is None, and it throws exceptions)

there is the test called for testing connection of psycopg and psycopg2 `test_connect_psycopg_multiple_hosts`